src/reason_parser.mly

```diff
@@ -2283,7 +2273,10 @@ value_type:
       { $1, Immutable, Concrete, $3 }
 ;
 constrain:
-        core_type EQUAL core_type          { $1, $3, symbol_rloc() }
+  | core_type EQUAL core_type {
+    let loc = mklocation $symbolstartpos $endpos in
+    $1, $3, loc
```

Comment:
If you can remember to, I suggest forming tuples like `($1, $3, loc)` to make it more intuitive for noobs to understand what's going on (slightly closer to the Reason syntax too).